### PR TITLE
Log when database migrations are going to happen, and ensure other reaper instances are stopped

### PR DESCRIPTION
**Pull request by michaelsembwever**

### CODE REVIEW

File: src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java [hunk 1]
```diff
@@ -201,18 +202,36 @@ private static void initializeAndUpgradeSchema(
...
+      if (4 <= currentVersion) {
+        List<String> otherRunningReapers = session.execute("SELECT reaper_instance_host FROM running_reapers").all()
+            .stream()
+            .map((row) -> row.getString("reaper_instance_host"))
+            .filter((reaperInstanceHost) -> !AppContext.REAPER_INSTANCE_ADDRESS.equals(reaperInstanceHost))
+            .collect(Collectors.toList());
+
+        Preconditions.checkState(
+            otherRunningReapers.isEmpty(),
+            "Database migration can not happen with other reaper instances running. Found ",
```
Comment (ossarga) [inline src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java:219]: nitpick: Consider rewording this log message to remove the "not":

> Unable to migrate database while other reaper instances are running. Found

**Review by ossarga — APPROVED** [review]
Changes LGTM.

I was able to test this by creating a docker environment with Cassandra 3.11.2, Reaper 0.7.0 and Reaper 1.2.3-SNAPSHOT. After starting Cassandra and Reaper 0.7.0, I started Reaper 1.2.3-SNAPSHOT and it halted prior to the database migration with the following message
```
java -jar cassandra-reaper-1.2.3-SNAPSHOT.jar server cassandra-reaper-cassandra.yaml
INFO   [2018-08-07 04:42:17,234] [main] o.e.j.u.log - Logging initialized @1700ms to org.eclipse.jetty.util.log.Slf4jLog
INFO   [2018-08-07 04:42:17,425] [main] i.d.s.DefaultServerFactory - Registering jersey handler with root path prefix: /
INFO   [2018-08-07 04:42:17,427] [main] i.d.s.DefaultServerFactory - Registering admin handler with root path prefix: /
INFO   [2018-08-07 04:42:17,427] [main] i.d.a.AssetsBundle - Registering AssetBundle with name: assets for path /webui/*
INFO   [2018-08-07 04:42:17,484] [main] i.c.ReaperApplication - initializing runner thread pool with 15 threads
INFO   [2018-08-07 04:42:17,500] [main] i.c.ReaperApplication - initializing storage of type: cassandra
INFO   [2018-08-07 04:42:17,522] [main] c.d.d.core - DataStax Java driver 3.5.0 for Apache Cassandra
INFO   [2018-08-07 04:42:17,534] [main] c.d.d.c.GuavaCompatibility - Detected Guava >= 19 in the classpath, using modern compatibility layer
INFO   [2018-08-07 04:42:17,752] [main] c.d.d.c.ClockFactory - Using native clock to generate timestamps.
INFO   [2018-08-07 04:42:17,952] [main] c.d.d.c.NettyUtil - Found Netty's native epoll transport in the classpath, using it
INFO   [2018-08-07 04:42:18,348] [main] c.d.d.c.p.DCAwareRoundRobinPolicy - Using data-center name 'datacenter1' for DCAwareRoundRobinPolicy (if this is incorrect, please provide the correct datacenter name with DCAwareRoundRobinPolicy constructor)
INFO   [2018-08-07 04:42:18,349] [main] c.d.d.c.Cluster - New Cassandra host cassandra/172.30.0.2:9042 added
INFO   [2018-08-07 04:42:18,623] [main] o.c.c.m.MigrationRepository - Found 16 migration scripts
WARN   [2018-08-07 04:42:18,624] [main] i.c.s.CassandraStorage - Starting db migration from 4 to 16…
Database migration can not happen with other reaper instances running. Found  [[172.30.0.3]]
``` 

I then stopped Reaper 0.7.0 and removed its entry from the `reaper_db.running_reapers` table. After starting Reaper 1.2.3-SNAPSHOT again it ran successfully with the following output.
```
java -jar cassandra-reaper-1.2.3-SNAPSHOT.jar server cassandra-reaper-cassandra.yaml
INFO   [2018-08-07 04:53:03,115] [main] o.e.j.u.log - Logging initialized @1635ms to org.eclipse.jetty.util.log.Slf4jLog
INFO   [2018-08-07 04:53:03,283] [main] i.d.s.DefaultServerFactory - Registering jersey handler with root path prefix: /
INFO   [2018-08-07 04:53:03,285] [main] i.d.s.DefaultServerFactory - Registering admin handler with root path prefix: /
INFO   [2018-08-07 04:53:03,285] [main] i.d.a.AssetsBundle - Registering AssetBundle with name: assets for path /webui/*
INFO   [2018-08-07 04:53:03,342] [main] i.c.ReaperApplication - initializing runner thread pool with 15 threads
INFO   [2018-08-07 04:53:03,358] [main] i.c.ReaperApplication - initializing storage of type: cassandra
INFO   [2018-08-07 04:53:03,385] [main] c.d.d.core - DataStax Java driver 3.5.0 for Apache Cassandra
INFO   [2018-08-07 04:53:03,391] [main] c.d.d.c.GuavaCompatibility - Detected Guava >= 19 in the classpath, using modern compatibility layer
INFO   [2018-08-07 04:53:03,597] [main] c.d.d.c.ClockFactory - Using native clock to generate timestamps.
INFO   [2018-08-07 04:53:03,787] [main] c.d.d.c.NettyUtil - Found Netty's native epoll transport in the classpath, using it
INFO   [2018-08-07 04:53:04,122] [main] c.d.d.c.p.DCAwareRoundRobinPolicy - Using data-center name 'datacenter1' for DCAwareRoundRobinPolicy (if this is incorrect, please provide the correct datacenter name with DCAwareRoundRobinPolicy
constructor)
INFO   [2018-08-07 04:53:04,123] [main] c.d.d.c.Cluster - New Cassandra host cassandra/172.30.0.2:9042 added
INFO   [2018-08-07 04:53:04,334] [main] o.c.c.m.MigrationRepository - Found 16 migration scripts
WARN   [2018-08-07 04:53:04,334] [main] i.c.s.CassandraStorage - Starting db migration from 4 to 16…
WARN   [2018-08-07 04:53:04,340] [main] i.c.s.c.Migration009 - Removing NULLs in the repair_run table. This may take some minutes…
WARN   [2018-08-07 04:53:04,349] [main] i.c.s.c.Migration009 - Removal of NULLs in the repair_run table completed.
INFO   [2018-08-07 04:53:09,320] [main] o.c.c.m.MigrationTask - Migrated keyspace reaper_db to version 16
WARN   [2018-08-07 04:53:09,338] [main] i.c.s.c.Migration016 - altering every table to set `dclocal_read_repair_chance` to zero…
WARN   [2018-08-07 04:53:09,349] [main] i.c.s.c.Migration016 - alter every table to set `dclocal_read_repair_chance` to zero completed.
INFO   [2018-08-07 04:53:09,660] [main] i.c.ReaperApplication - no JMX connection factory given in context, creating default
INFO   [2018-08-07 04:53:09,682] [main] i.c.ReaperApplication - creating and registering health checks
INFO   [2018-08-07 04:53:09,682] [main] i.c.ReaperApplication - creating resources and registering endpoints
INFO   [2018-08-07 04:53:10,729] [main] i.c.s.SchedulingManager - Starting new SchedulingManager instance
INFO   [2018-08-07 04:53:10,732] [main] i.c.ReaperApplication - Initializing JMX seed list for all clusters...
INFO   [2018-08-07 04:53:10,741] [main] i.c.ReaperApplication - Initialized JMX seed list for all clusters.
INFO   [2018-08-07 04:53:10,741] [main] i.c.ReaperApplication - resuming pending repair runs
INFO   [2018-08-07 04:53:10,747] [main] i.c.ReaperApplication - Initialization complete!
WARN   [2018-08-07 04:53:10,747] [main] i.c.ReaperApplication - Reaper is ready to get things done!
INFO   [2018-08-07 04:53:10,747] [main] i.d.s.DefaultServerFactory - Registering jersey handler with root path prefix: /
INFO   [2018-08-07 04:53:10,747] [main] i.d.s.DefaultServerFactory - Registering admin handler with root path prefix: /
INFO   [2018-08-07 04:53:10,768] [main] i.d.s.ServerFactory - Starting cassandra-reaper
```

**Comment by michaelsembwever** [comment]
>  removed its entry from the reaper_db.running_reapers table.

Hopefully that step is not required.

File: src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java (hunk 1 above)
Comment (michaelsembwever) [reply src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java:219]: bugger, i missed this.

**Comment by adejanovski** [comment]
Since the `running_reapers` table has a TTL of 3 minutes, this means that it's not possible to perform an upgrade until 3 minutes after the last reaper has been stopped. 
Might be worth mentioning in the error message.

**Pull request merged**